### PR TITLE
Icons for Google, Facebook and Apple

### PR DESCRIPTION
**Pull request by oliverlloyd**
## What is the purpose of this change?
Adds icons for Google, Apple and Facebook that have the brand colours for each of these companies.

These icons are being promoted from the identity gateway project, see: https://github.com/guardian/gateway/pull/553

### Example usage
Screenshot 2021-05-25 at 13 37 56

### CODE REVIEW

**Comment by oliverlloyd** [comment]
[Social icons in Source](https://trello.com/c/chWHkFNQ/2994-social-icons-in-source)

File: src/core/icons/apple-brand.tsx [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+			<path
+				d="M15.1173 10.7533C15.1173 13.7106 18.1328 14.8684 18.1328 14.8684C17.4932 16.9092 16.2099 18.6073 14.3429 19.6658C13.0163 20.4178 11.4836 19.6658 9.85097 19.6658C8.21829 19.6658 6.68561 20.4178 5.35908 19.6658C2.58487 18.093 0.867188 14.1765 0.867188 10.7533C0.867188 7.77471 1.46602 6.31155 3.39656 5.31825C5.32711 4.32495 7.77481 5.9394 9.85097 5.9394C11.9271 5.9394 14.7709 4.14886 16.3054 5.31822C16.9019 5.77284 17.3439 6.23053 17.6712 6.7158C17.6712 6.7158 15.1173 7.79601 15.1173 10.7533Z"
+				fill="black"
+			/>
+			<path
+				d="M10.9888 2.23343C10.1198 3.4648 9.52522 5.80437 9.52522 5.80437C9.52522 5.80437 11.823 4.76878 12.6033 3.48044C13.3212 2.29499 13.2966 1.61225e-05 13.2966 1.61225e-05C13.2966 1.61225e-05 11.7119 1.20871 10.9888 2.23343Z"
+				fill="black"
+			/>
+		</svg>
+	);
```
Comment (mxdvl) [inline src/core/icons/apple-brand.tsx:19]: From https://appleid.cdn-apple.com/appleauth/static/jsapi/appleid/1/en_US/appleid.auth.js

I could extract 

```html
<svg xmlns="http://www.w3.org/2000/svg" style="overflow:visible" width="100%" height="100%" viewBox="6 6 44 44">
	<g fill="none" fill-rule="evenodd">
		<path fill-rule="nonzero"
			d="M28.2226562,20.3846154 C29.0546875,20.3846154 30.0976562,19.8048315 30.71875,19.0317864 C31.28125,18.3312142 31.6914062,17.352829 31.6914062,16.3744437 C31.6914062,16.2415766 31.6796875,16.1087095 31.65625,16 C30.7304687,16.0362365 29.6171875,16.640178 28.9492187,17.4494596 C28.421875,18.06548 27.9414062,19.0317864 27.9414062,20.0222505 C27.9414062,20.1671964 27.9648438,20.3121424 27.9765625,20.3604577 C28.0351562,20.3725366 28.1289062,20.3846154 28.2226562,20.3846154 Z M25.2929688,35 C26.4296875,35 26.9335938,34.214876 28.3515625,34.214876 C29.7929688,34.214876 30.109375,34.9758423 31.375,34.9758423 C32.6171875,34.9758423 33.4492188,33.792117 34.234375,32.6325493 C35.1132812,31.3038779 35.4765625,29.9993643 35.5,29.9389701 C35.4179688,29.9148125 33.0390625,28.9122695 33.0390625,26.0979021 C33.0390625,23.6579784 34.9140625,22.5588048 35.0195312,22.474253 C33.7773438,20.6382708 31.890625,20.5899555 31.375,20.5899555 C29.9804688,20.5899555 28.84375,21.4596313 28.1289062,21.4596313 C27.3554688,21.4596313 26.3359375,20.6382708 25.1289062,20.6382708 C22.8320312,20.6382708 20.5,22.5950413 20.5,26.2911634 C20.5,28.5861411 21.3671875,31.013986 22.4335938,32.5842339 C23.3476562,33.9129053 24.1445312,35 25.2929688,35 Z">
		</path>
	</g>
</svg>
```

**Review by mxdvl — COMMENTED** [review]
Should we really defer to these companies' colours? It looks really jarring inside the blue border.

File: src/core/icons/apple-brand.tsx (hunk 1 above)
Comment (oliverlloyd) [reply src/core/icons/apple-brand.tsx:19]: > I think it's worth keeping a track of how and where we got these icons from

+1 - I'll try to dig out refs to the source

**Comment by oliverlloyd** [comment]
> Should we really defer to these companies' colours

When I checked this with Ana F, she said that they had done user testing with both variations, buttons with brand colours and plain black buttons. The coloured variation performed better so a design call was made to favour those.

File: src/core/icons/apple-brand.tsx (hunk 1 above)
Comment (mxdvl) [reply src/core/icons/apple-brand.tsx:19]: Cross-posting from the channel, Wikimedia Commons:

https://commons.wikimedia.org/wiki/File:Apple_logo_black.svg

https://commons.wikimedia.org/wiki/File:Google_%22G%22_Logo.svg

https://commons.wikimedia.org/wiki/File:Facebook_f_logo_(2019).svg

**Comment by oliverlloyd** [comment]
> looks really jarring inside the blue border

Just an update on this. That light blue border has been decided against in favour of brand.400 dark blue

**Comment by SiAdcock** [comment]
Thanks for rolling this forward @oliverlloyd . Would it be possible to update the names of the files and components to use the word "brand" as a suffix?
i.e.
- `SvgAppleBrand`
- `apple-brand.tsx`
- `SvgFacebookBrand`
- `facebook-brand.tsx`
- `SvgGoogleBrand`
- `google-brand.tsx`

**Comment by oliverlloyd** [comment]
> Would it be possible to update the names of the files and components to use the word "brand" as a suffix?

This is now done

File: src/core/icons/apple-brand.tsx (hunk 1 above)
Comment (mxdvl) [reply src/core/icons/apple-brand.tsx]: I think we should update the Apple logo with the one I posted above, as it currently just looks odd, and has undergone too severe an optimisation.

Comment (oliverlloyd) [reply src/core/icons/apple-brand.tsx]: I did try the svg above but it didn't appear consistently with the icons around it. I tried tweaking the sizes and viewbox but I'm not really sure what I'm doing at that point

Screenshot 2021-05-27 at 09 34 52
Screenshot 2021-05-27 at 09 35 01

Comment (mxdvl) [reply src/core/icons/apple-brand.tsx]: Fixed in 1912a7f

**Comment by oliverlloyd** [comment]
🚢 ⛵ 🚀 ?

**Review by SiAdcock — APPROVED** [review]
Looks awesome, thanks so much @oliverlloyd and @mxdvl 🚀

**Comment by mxdvl** [comment]
Worth noting that the brand icon for Apple on a dark blue background has a very poor contrast ratio.

poor contrast

**Comment by SiAdcock** [comment]
> very poor contrast ratio

Yes, it's horrifying! Hopefully no designer would ever hand over a mockup like this 😄

**Pull request merged**